test/junit/scala/collection/immutable/ListSetTest.scala

```diff
@@ -50,4 +52,67 @@ class ListSetTest {
     val s = ListSet(1, 2, 3, 5, 4)
     assertEquals(List(1, 2, 3, 5, 4), s.iterator.toList)
   }
+
```

Comment:
I brought these tests forward, but dropped the change to the implementation which was a fix for an optimization that itself is 2.12.x only.